### PR TITLE
Bugfix: fsnotify leak

### LINKED ISSUE #76949: PluginWatcher:  if do not close watcher when timeout is expected?
**What happened**:

https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/util/pluginwatcher/plugin_watcher.go#L160

When timeout happens, we should not close watcher? Is this desired?

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by cwdsuzhou**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

Bugfix: fsnotify leak

**Which issue(s) this PR fixes**:

Fixes #76949  

**Special notes for your reviewer**:

/cc @mattjmcnaughton 

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by mattjmcnaughton — APPROVED** [review]
/lgtm
/ok-to-test

thanks for this pr :)

**Comment by Random-Liu** [comment]
I think close fsnotify after fully stopped seems correct to me.

The problem is that:
1) Why `Stop` is not idempotent? The second `Stop` will try closing a closed channel, which will cause a panic.
1) Why we don't retry if  `Stop` returns error.

**Comment by cwdsuzhou** [comment]
> I think close fsnotify after fully stopped seems correct to me.
> 
> The problem is that:
> 
> 1. Why `Stop` is not idempotent? The second `Stop` will try closing a closed channel, which will cause a panic.
> 2. Why we don't retry if  `Stop` returns error.

Yes, close fsnotify after fully stopped seems correct. However,  when Close time out, fsnotify would leak and have not any retry.

File: pkg/kubelet/util/pluginwatcher/plugin_watcher.go [hunk 1]
```diff
@@ -166,14 +166,14 @@ func (w *Watcher) Stop() error {
 		w.wg.Wait()
 	}()
 
+	defer w.fsWatcher.Close()
+
 	select {
 	case <-c:
 	case <-time.After(11 * time.Second):
```
Comment (cwdsuzhou) [inline pkg/kubelet/util/pluginwatcher/plugin_watcher.go]: I found the error return is never used expect in tests

**Comment by cwdsuzhou** [comment]
> I think close fsnotify after fully stopped seems correct to me.
> 
> The problem is that:
> 
> 1. Why `Stop` is not idempotent? The second `Stop` will try closing a closed channel, which will cause a panic.
> 2. Why we don't retry if  `Stop` returns error.

I would try to fix the 2 points later, thanks

**Comment by cwdsuzhou** [comment]
/close

see https://github.com/kubernetes/kubernetes/pull/77907

**Pull request closed without merging**